elixir/apps/domain/test/domain/auth/adapters/google_workspace/jobs/sync_directory_test.exs

```diff
@@ -65,14 +65,12 @@ defmodule Domain.Auth.Adapters.GoogleWorkspace.Jobs.SyncDirectoryTest do
       end)
 
       GoogleWorkspaceDirectory.override_endpoint_url("http://localhost:#{bypass.port}/")
-      GoogleWorkspaceDirectory.mock_groups_list_endpoint(bypass, [])
-      GoogleWorkspaceDirectory.mock_organization_units_list_endpoint(bypass, [])
-      GoogleWorkspaceDirectory.mock_users_list_endpoint(bypass, [])
 
       {:ok, pid} = Task.Supervisor.start_link()
+
       assert execute(%{task_supervisor: pid}) == :ok
 
-      assert_receive {:bypass_request,
+      refute_receive {:bypass_request,
```

Comment:
I know sometimes `refute_receive` can be tricky to rely on. Would it make sense to add another assertion here to make sure this does what we want?

Like `assert_receive {:bypass_request, ...}` but with the data we want?

Reply:
I definitely agree with you, however this test gave me a little bit of a headache and this was the best I could come up with, without having to do a large refactor. 

The difficulty with this test is that we no longer send any request if we don't get an access token through our Service Account key.  Before this PR this test worked well, because we always fell back to the personal access token and we could check that a request was sent with that personal access token.

With the changes I've made, there's no further request sent and this `execute` function always ends with an `:ok`, so I'm not sure what we could `assert` on.

If there's something you see that I may have missed I'm definitely happy to update the test.

Reply:
👍